File: sol/lab00/cpp/task-1/sol1_sort_desc.cpp
// SPDX-License-Identifier: BSD-3-Clause

#include <algorithm>
#include <vector>
using namespace std;

class Solution {
public:
    // Solutie: Sortam vectorul in ordine descrescatoare si returnam al k-lea cel mai mare element.
    // Complexitate: O(nlogn)
    int findKthLargest(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end(), greater<int>());
        return nums[k - 1];
    }
};

File: sol/lab00/java/task-1/sol1_sort_desc.java
// SPDX-License-Identifier: BSD-3-Clause

import java.util.Arrays;
import java.util.Collections;

class Solution {
    // Solutie: Sortam vectorul in ordine descrescatoare si returnam al k-lea cel mai mare element.
    // Complexitate: O(nlogn)
    public int findKthLargest(int[] nums, int k) {
        Integer[] numsBoxed = Arrays.stream(nums).boxed().toArray(Integer[]::new);
        Arrays.sort(numsBoxed, Collections.reverseOrder());
        return numsBoxed[k - 1];
    }
}
